modules/search/class-jetpack-instant-search.php

```diff
@@ -575,7 +576,9 @@ public function auto_config_excluded_post_types() {
 			}
 		}
 
-		$post_types_to_disable = array_diff( $post_types, $enabled_post_types );
-		update_option( Jetpack_Search_Options::OPTION_PREFIX . 'excluded_post_types', join( ',', $post_types_to_disable ) );
+		if ( ! empty( $enabled_post_types ) ) {
```

Comment:
We only now do this step if we found some enabled post types in the widgets.

Previously, this would have disabled all post types when there were no enabled post types in the widgets:

```
$post_types = [ 'post', 'page'];
$enabled_post_types = [];
$post_types_to_disable = array_diff( $post_types, $enabled_post_types); // returns ['post', 'page']
```